File: Backend/src/main/java/com/resumebuilder/backend/dto/ProjectDTO.java
package com.resumebuilder.backend.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import jakarta.validation.constraints.NotBlank;
import java.time.LocalDate;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ProjectDTO {
    private Long id;

    @NotBlank(message = "Project title is required")
    private String title;

    private String link;
    private LocalDate date;
    private String techStack;
    private List<String> bulletPoints;
}

File: app/src/main/java/com/snehil/cvoptima/data/remote/model/ProjectDto.kt
package com.snehil.cvoptima.data.remote.model

import com.google.gson.annotations.SerializedName

data class ProjectDto(
    @SerializedName("id") val id: Long? = null,
    @SerializedName("title") val title: String,
    @SerializedName("link") val link: String? = null,
    @SerializedName("date") val date: String? = null,
    @SerializedName("techStack") val techStack: String? = null,
    @SerializedName("bulletPoints") val bulletPoints: List<String>? = null
)
